Doc/library/asyncio-task.rst

```diff
@@ -1153,6 +1153,11 @@ Task Object
 
       Return the coroutine object wrapped by the :class:`Task`.
 
+      .. note::
+
+        This will return ``None`` for Tasks which have already
+        completed eagerly.
```

Comment:
I feel this requires a link to the eager task factory stuff.

Reply:
Good idea.

Reply:
Resolved in the new PR https://github.com/python/cpython/pull/104304